File: frontend/src/app/dtos/user/login.dto.ts
import {IsString, IsNotEmpty, IsPhoneNumber, IsDate} from 'class-validator'

export class LoginDto{
    @IsString()
    @IsNotEmpty()
    password: string;

    @IsPhoneNumber()
    phoneNumber:string;

  constructor(data:any){
    this.phoneNumber=data.phoneNumber;
    this.password=data.password;
  }
}

File: android/lib/dtos/login_dto.dart
class LoginDto{
    final String password;
    final String phoneNumber;

    LoginDto({
      required this.phoneNumber,
      required this.password,
    });

    Map<String, dynamic> toJson() {
      return {
        'phoneNumber': phoneNumber,
        'password': password,
      };
    }
}